src/Neo/SmartContract/Native/CryptoLib.cs

```diff
@@ -115,5 +117,34 @@ public static bool VerifyWithECDsaV0(byte[] message, byte[] pubkey, byte[] signa
                 return false;
             }
         }
+
+        /// <summary>
+        /// Verifies that a digital signature is appropriate for the provided key and message using the Ed25519 algorithm.
+        /// </summary>
+        /// <param name="message">The signed message.</param>
+        /// <param name="publicKey">The Ed25519 public key to be used.</param>
+        /// <param name="signature">The signature to be verified.</param>
+        /// <returns><see langword="true"/> if the signature is valid; otherwise, <see langword="false"/>.</returns>
+        [ContractMethod(Hardfork.HF_Echidna, CpuFee = 1 << 15)]
+        public static bool VerifyWithEd25519(byte[] message, byte[] publicKey, byte[] signature)
+        {
+            if (signature.Length != Ed25519.SignatureSize)
+                return false;
+
+            if (publicKey.Length != Ed25519.PublicKeySize)
+                return false;
+
+            try
+            {
+                var verifier = new Ed25519Signer();
+                verifier.Init(false, new Ed25519PublicKeyParameters(publicKey, 0));
+                verifier.BlockUpdate(message, 0, message.Length);
+                return verifier.VerifySignature(signature);
+            }
+            catch (Exception)
```

Comment:
Well, I was talking only about ArgumentException, e.g. only about input data of invalid length and some other ArgumentException thrown by this part of code. However, catching any exception is a valid solution as far.

@roman-khimov, what do you think about cases when we should return true/false or FAULT VM?

Reply:
Can be both ways, but we need to know all of the exceptions that can happen in this block. Some are likely OK to be converted to `false` result (invalid signature), some may not (`verifier` constructor failure?). I'd expect some symmetry to ECDSA verification function. Like what happens if the key is wrong?